### PR TITLE
cadence: refactor polling_request initialization

### PR DESCRIPTION
**Pull request by aleh-null**
The loop is very confusing and hard to follow for this reason a minor refactoring is done. It should also help us to get a better understanding of cli. The error and logging issues are somewhere within this routine.

### CODE REVIEW

File: cadence/activity_loop.py [hunk 1]
```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
...
-                if err:
-                    logger.error("PollForActivityTask failed: %s", err)
-                    continue
-                # TODO: move this one to handle_task() or check 
-                elif not task.task_token:
-                    logger.debug("PollForActivityTask has no task_token (expected): %s", task)
-                    continue
-                logger.debug("PollForActivityTask: done")
+                task = service.poll_for_activity_task(polling_request)
+            except RuntimeError as ex:
```
Comment (aleh-null) [inline cadence/activity_loop.py]: Maybe it's a good idea to use one catch, does not really matter if it's runtime err or not.

File: cadence/activity_loop.py [hunk 2]
```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
...
                     task_token=task.task_token,
-                    result = json.dumps(ret),
-                    identity = WorkflowService.get_identity(),
+                    result=json.dumps(ret),
+                    identity=WorkflowService.get_identity(),
                 )
-                _, err = service.respond_activity_task_completed(respond)
-                if err:
-                    logger.error("Error invoking RespondActivityTaskCompleted: %s", error)
+                service.respond_activity_task_completed(response)
```
Comment (aleh-null) [inline cadence/activity_loop.py]: do we need try/catch here?

File: cadence/activity_loop.py [hunk 3]
```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
...
+        logger.fatal(f"activity_task_loop: Uncought exception in main loop.",exec_info=True)
     finally:
         worker.notify_thread_stopped()
 
 
-def handle_task(task, activities):
+
+
+
+def handle_task(task, activities) -> Dict:
```
Comment (aleh-null) [inline cadence/activity_loop.py]: Can we set types for args here for linting?

File: cadence/activity_loop.py [hunk 4]
```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
...
+    def handle_error(msg: str, exception: Type[Exception]) -> NoReturn:
+        logger.error(msg)
+        raise exception(msg)
+
     # TODO: finish this one
```
Comment (aleh-null) [inline cadence/activity_loop.py]: Remove TODO

File: cadence/activity_loop.py [hunk 5]
```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
...
-def handle_task_err(task):
-    return
+    if not task.task_token:
+        handle_error("task.task_token was not provided, but is expected.",
+                     RuntimeError)
+    if task.activity_type.name not in activities:
+        handle_error(
+            f"handle_task: Activity type {task.activity_type.name} not found",
+            RuntimeError)
+    fn = activities[task.activity_type.name]
```
Comment (aleh-null) [inline cadence/activity_loop.py]: move it to line before 107

File: cadence/activity_loop.py (hunk 1 above)
Comment (chm90) [reply cadence/activity_loop.py]: Agree

File: cadence/activity_loop.py (hunk 2 above)
Comment (chm90) [reply cadence/activity_loop.py]: NO

File: cadence/activity_loop.py (hunk 3 above)
Comment (chm90) [reply cadence/activity_loop.py]: yes

File: cadence/activity_loop.py (hunk 5 above)
Comment (chm90) [reply cadence/activity_loop.py]: We just checked that it exists the lines before. I think it's more clear if it is right when we have verified that it exists

File: cadence/activity_loop.py (hunk 1 above)
Comment (aleh-null) [reply cadence/activity_loop.py]: Delete this one?

Comment (chm90) [reply cadence/activity_loop.py]: should be?

File: cadence/activity_loop.py (hunk 5 above)
Comment (aleh-null) [reply cadence/activity_loop.py]: sure!
